src/mscorlib/src/System/Array.cs

```diff
@@ -2502,45 +2501,43 @@ sealed class SZArrayHelper {
         // This is a normal generic Enumerator for SZ arrays. It doesn't have any of the "this" stuff
         // that SZArrayHelper does.
         //
-        [Serializable] private sealed class SZGenericArrayEnumerator<T> : IEnumerator<T> {
+        [Serializable]
+        private sealed class SZGenericArrayEnumerator<T> : IEnumerator<T>
+        {
             private T[] _array;
             private int _index;
-            private int _endIndex; // cache array length, since it's a little slow.
```

Comment:
@stephentoub The code was likely copied from here: https://github.com/jamesqo/coreclr/blob/d613bf15ff7b8771125db49841868333f6b4bf5b/src/mscorlib/src/System/Array.cs#L2238 For non-generic arrays `Length` seems to compile into an actual call rather than a pointer access:

```asm
; Assembly listing for method Program:Foo(ref)
; Emitting BLENDED_CODE for X64 CPU with SSE2
; optimized code
; rsp based frame
; partially interruptible
; Final local variable assignments
;
;  V00 arg0         [V00,T00] (  5,   4  )     ref  ->  rcx
;  V01 OutArgs      [V01    ] (  1,   1  )  lclBlk (32) [rsp+0x00]
;
; Lcl frame size = 40

G_M17832_IG01:
       4883EC28             sub      rsp, 40

G_M17832_IG02:
       4885C9               test     rcx, rcx
       740C                 je       SHORT G_M17832_IG03
       E84AC0CB5F           call     Array:get_Length():int:this
       8BC8                 mov      ecx, eax
       E8ABFBFFFF           call     Console:WriteLine(int)

G_M17832_IG03:
       90                   nop

G_M17832_IG04:
       4883C428             add      rsp, 40
       C3                   ret

; Total bytes of code 27, prolog size 4 for method Program:Foo(ref)
; ============================================================
; Assembly listing for method Program:Bar(ref)
; Emitting BLENDED_CODE for X64 CPU with SSE2
; optimized code
; rsp based frame
; partially interruptible
; Final local variable assignments
;
;  V00 arg0         [V00,T00] (  4,   3.5)     ref  ->  rcx
;  V01 OutArgs      [V01    ] (  1,   1  )  lclBlk (32) [rsp+0x00]
;
; Lcl frame size = 40

G_M17187_IG01:
       4883EC28             sub      rsp, 40

G_M17187_IG02:
       4885C9               test     rcx, rcx
       7408                 je       SHORT G_M17187_IG03
       8B4908               mov      ecx, dword ptr [rcx+8]
       E87FFBFFFF           call     Console:WriteLine(int)

G_M17187_IG03:
       90                   nop

G_M17187_IG04:
       4883C428             add      rsp, 40
       C3                   ret

; Total bytes of code 23, prolog size 4 for method Program:Bar(ref)
; ============================================================

```